lms/djangoapps/instructor/views/tools.py

```diff
@@ -204,6 +204,16 @@ def visit(node):
         else:
             api.set_date_for_block(course.id, block.location, 'due', None, user=student, reason=reason, actor=actor)
 
+    # edx-proctoring is checking cached course dates, so the overrides made above will not be enforced until the
+    # TieredCache is reloaded. This can lead to situations when a student's extension is revoked, but they can still
+    # complete an exam for some time. The instructors don't have a way of checking whether the cache has been
+    # regenerated because the Instructor Dashboard simply lists all extensions. Therefore, to avoid having a confusing
+    # user experience, we want trigger cache regeneration after changing the due date.
+    api.get_dates_for_course(course.id, user=student, published_version=version, use_cached=False)
+    if version:
+        # edx-proctoring is not using the course version while checking its dates.
+        api.get_dates_for_course(course.id, user=student, use_cached=False)
```

Comment:
Sorry @Agrendalath I don't quite understand this check here. If `version` is set then why do we need to reset the cache without it? Doesn't it negate the previous step (line 212)?

Reply:
@pkulkark,

Some context: there are two instances of this cache:
1. The one without the version.
2. The one with the version (optional).

By "instances", I mean that they are using different caching keys. 

`edx-proctoring` does not handle the versioning, so it retrieves exam dates [without specifying the published version](https://github.com/openedx/edx-proctoring/blob/0ae746c253db5910cd6f91abf60e19771102130c/edx_proctoring/utils.py#L279) from `edx-when`. `edx-when` uses `None` as a default `published_version` for [this query](https://github.com/openedx/edx-when/blob/0254185d1cd51e2338ab7e60b5b4fce5ac78b580/edx_when/api.py#LL276C5-L276C17). Therefore, we are using only the first instance of the cache in this scenario. But for consistency, we need to regenerate the second one too, if it exists.

Now, back to the code in this PR:
If the version is **not** set, we are resetting the cache without the version in line 212. Therefore, we don't need to run this again in line 215.
However, when the version is set, then line 212 resets only the second instance of the cache. The first one (without the version) is intact, so we need to reset it too.